### ISSUE TITLE
HAProxy doesn't create the default_backend rule when needed

### ISSUE THREAD

**Issue by wouterd** [tutumcloud/haproxy#125]
I have the following situation:

One NGinx should handle all traffic except for the traffic to api.part-up.com (jvm based rest api).

I have no VIRTUAL_HOST configured on the nginx container and VIRTUAL_HOST=api.part-up.com on the API container.

When I start the HAProxy, it does create the default_service, but it doesn't add the default_backend rule, the culprit seems the if-statement here: https://github.com/tutumcloud/haproxy/blob/master/haproxy/haproxy.py#L414

Probably best to just remove that if, but it might be there for a reason. :)

**Comment by tifayuki** [comment]
Yes, there is something wrong when some of the linked container have `VIRTUAL_HOST` while some don't.

If VIRTUAL_HOST is set in any of the linked containers, the default frontend is not added. It is on purpose, as It is meaningless to bind a port `80` twice, which could become a problem that the default frontend will take effect but the one with ACL rules is ignored.

In the backends part, all the routes without `VIRTUAL_HOST` set are added to a default backend, and it is useless because there is no corresponding rule to that. 

Currently, the behavior of this image is that if `VIRTUAL_HOST` is set in one of the containers, the containers without `VIRTUAL_HOST` are ignored. This is the same as my design: `VIRUTAL_HOST`, all or none.

In the future, I may try to add a default ACL rule to port 80's frontend. But for the moment, to your use case, you need to set VIRUTAL_HOST="http://*, https://*", and "VIRTUAL_HOST_WEIGHT=1" in you nginx container.

Thank you for pointing out this issue.
